Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,263 @@
+        {
+            return Validate(new List<JObject> { SerializeToJObject(schema) }, rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas in Dictionary with the given rules.
+        /// </summary>
+        /// <param name="schemas">The schemas to be validated</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>        
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaAsForgeTreeDictionary(Dictionary<string, ForgeTree> schemas, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            return Validate(ConvertDictionaryToJObjectList(schemas, validateAsDictionary), rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas in string with the given rules.
+        /// </summary>
+        /// <param name="schema">The schema to be validated</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaAsString(string schema, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemaList = ConvertStringToJObjectList(schema, validateAsDictionary);
+
+            return Validate(schemaList, rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas from input path with the given rules.        
+        /// </summary>
+        /// <param name="path">The path that contains a schema file</param>
+        /// <param name="rules">The rules used to validate input schemas is only allowed in string or JSchema type</param>
+        /// <param name="validateAsDictionary">True if the custom rules is to handle the whole dictionary</param>
+        /// <returns>The result of schema validation. The errorList would contain error message if validation fails</returns>
+        public static bool ValidateSchemaFromPath(string path, object rules, bool validateAsDictionary, out IList<ValidationError> errorList)
+        {
+            List<JObject> schemas = GetSchemaFromPath(path, validateAsDictionary);
+
+            return Validate(schemas, rules, out errorList);
+        }
+
+        /// <summary>
+        /// Validates single or multiple schemas from input path with the given rules.
```

Comment:
>Validates single or multiple schemas from input path with the given rules. [](http://example.com/codeflow?start=12&length=74)

"Validates single or multiple schemas from all files under the directory path, with the given rules."